### PR TITLE
[poco] bin and debug/bin directories should also be removed on macOS

### PR DESCRIPTION
**Pull request by angelmixu**
In macOS there's an error if both directories are created but empty.

### CODE REVIEW

**Comment by angelmixu** [comment]
It fails building windows on the PR-test, but I can't take a look at what's failing (or I don't know how).
It's strange, as the change I made only affects macOS :/
Anyone with suggestions?

**Comment by JackBoosY** [comment]
@angelmixu Please download and check failure log [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=27066&view=artifacts&type=publishedArtifacts).

**Comment by c72578** [comment]
[error] REGRESSION: qt5-winextras:x64-windows
```console
C:\vsts\_work\1\s\buildtrees\qt5-winextras\src\5.12.5-99fbd09e12\src\winextras\qwinfunctions.cpp(2045): 
fatal error C1051: program database file, 
'C:\vsts\_work\1\s\buildtrees\qt5-winextras\x64-windows-dbg\src\winextras\.obj\debug\Qt5WinExtrasd.vc.pdb', 
has an obsolete format, delete it and recompile
```

**Comment by angelmixu** [comment]
Thanks!
It looks like a bug in the qt5-winextras port, and not in this PR changes, isn't it?
The same error happens in https://github.com/microsoft/vcpkg/pull/9800 and https://github.com/microsoft/vcpkg/pull/9801 I can't get why this change is messing with pdb's :/

**Comment by JackBoosY** [comment]
@Neumann-A Can you confirm this?

Thanks.

File: ports/poco/portfile.cmake [hunk 1]
```diff
@@ -121,7 +121,7 @@ endif()
 
 
 #
-if (VCPKG_LIBRARY_LINKAGE STREQUAL static OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if (VCPKG_LIBRARY_LINKAGE STREQUAL static OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```
Comment (Neumann-A) [inline ports/poco/portfile.cmake]: should be replaced with newer variables like VCPKG_TARGET_IS_(OSX|LINUX)

**Comment by Neumann-A** [comment]
make a whitespace change to qt5-winextras and force a rebuild. I doubt that the changes to poco have anything to with the observed regression

**Comment by angelmixu** [comment]
Looks like this is working now! :D

**Comment by JackBoosY** [comment]
We need to test these features in all triplets.

File: ports/poco/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/poco/portfile.cmake]: @angelmixu Can you replace this line to `if (VCPKG_LIBRARY_LINKAGE STREQUAL static OR VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)`?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/poco/portfile.cmake (hunk 1 above)
Comment (angelmixu) [reply ports/poco/portfile.cmake]: sure! uploaded :)

**Comment by JackBoosY** [comment]
These regressions seem to be false, please wait for our processing to make changes later.

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!

**Pull request merged**